neurodamus/utils/memory.py

```diff
@@ -267,7 +269,37 @@ def display_total(self):
         logging.info("| {:<40s} | {:12.1f} |".format("Cells", self.cell_memory_total))
         logging.info("| {:<40s} | {:12.1f} |".format("Synapses", self.synapse_memory_total))
         logging.info("+{:-^57}+".format(""))
-        grand_total = full_overhead + self.cell_memory_total + self.synapse_memory_total
-        grand_total = pretty_printing_memory_mb(grand_total)
+        self.grand_total = full_overhead + self.cell_memory_total + self.synapse_memory_total
+        grand_total = pretty_printing_memory_mb(self.grand_total)
         logging.info("| {:<40s} | {:>12s} |".format("GRAND TOTAL", grand_total))
         logging.info("+{:-^57}+".format(""))
+
+    def total_memory_available():
+        """
+        Returns the total memory available in the system in MB
+        """
+        try:
+            virtual_memory = psutil.virtual_memory()
+            return virtual_memory.total / (1024 * 1024)  # Total available memory in MB
+        except Exception as e:
+            logging.error(f"Error: {e}")
+            return None
+
+    @run_only_rank0
+    def display_node_suggestions(self):
+        """
+        Display suggestions for how many nodes are approximately
+        necessary to run the simulation based on the memory available
+        on the current node.
+        """
+        node_total_memory = DryRunStats.total_memory_available()
+        if node_total_memory is None:
+            logging.warning("Unable to get the total memory available on the current node.")
+            return
```

Comment:
I don't agree with this. Neurodamus shouldn't integrate anything relative to BB5 since, being an OSS project, it would be a totally arbitrary limit to the program. 

BB5 will be decomissioned soon so making it specifically for this machine would be shortsighted IMHO if we imagine the program being used on other clusters.

We can discuss for sure improvements on how to evaluate the total memory but for the moment encouraging the user to run dry runs in the same kind of nodes where they intend to run the simulation seems to be a good start.

Reply:
Humm, good point it's open source now. We might then at some point provide options to override the destination node properties, but good for now